analyzere/extras/visualizations.py

```diff
@@ -0,0 +1,272 @@
+import hashlib
+
+from analyzere import LayerView
+from datetime import datetime
+from graphviz import Digraph
+from IPython.display import FileLink
```

Comment:
Not all users of this library will run this inside of an IPython/Jupyter environment. I'd suggest to conditionally import this by wrapping it into a `try` block. Inside of the `try` block you can set a module flag variable to indicate whether the execution is within IPython. You can later refer to the flag to provide different return values (see below).